src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +52,8 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
-    ssize_t len = recvmsg(sock, &msg, 0);
+    ssize_t len;
+    while (CheckInterrupted(len = recvmsg(sock, &msg, 0)));
```

Comment:
Nit: I really, really do not like the use of assignment as an expression. Would you mind changing all of these to do/while loops?